### PR TITLE
update law syntax

### PR DESCRIPTION
**Pull request by mlutze**
Fixes https://github.com/flix/flix/issues/1759

```flix
law myLaw: forall (_x: a) . true

law myLaw2: forall[b](_x: b) . true
```
is equivalent to
```flix
def myLaw(_x: a): Bool = true

def myLaw2[b](_x: b) = true
```

A lot ended up getting commented out, since there's some not-easily-automated refactoring to be done around existing laws.

### Features
* Update law syntax
  * [x] impl
  * [x] test

### General requirements
* [x] Don't make a complete mess
* [x] Don't break anything else
* [x] Provide documentation
* [x] Add license to new files

### CODE REVIEW

**Comment by mlutze** [comment]
@magnus-madsen RFR

File: main/test/ca/uwaterloo/flix/TestExamples.scala [hunk 1]
```diff
@@ -41,20 +41,21 @@ class TestExamples extends Suites(
...
-  new FlixTest("Parity", "examples/domains/Parity.flix", "examples/domains/Belnap.flix")(Options.TestWithLibrary),
-  new FlixTest("ParitySign", "examples/domains/ParitySign.flix", "examples/domains/Belnap.flix")(Options.TestWithLibrary),
-  new FlixTest("PrefixSuffix", "examples/domains/PrefixSuffix.flix", "examples/domains/Belnap.flix")(Options.TestWithLibrary),
-  new FlixTest("Sign", "examples/domains/Sign.flix", "examples/domains/Belnap.flix")(Options.TestWithLibrary),
-  new FlixTest("StrictSign", "examples/domains/StrictSign.flix", "examples/domains/Belnap.flix")(Options.TestWithLibrary),
+  // TODO re-enable these tests after their laws are update to the new syntax
+//  new FlixTest("TestBelnap", "examples/domains/Belnap.flix")(Options.TestWithLibrary),
+//  new FlixTest("TestConstant", "examples/domains/Constant.flix", "examples/domains/Belnap.flix")(Options.TestWithLibrary),
+//
+//  new FlixTest("ConstantParity", "examples/domains/ConstantParity.flix", "examples/domains/Belnap.flix")(Options.TestWithLibrary),
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/TestExamples.scala:48]: I don't want to disable these tests (for other reasons- we refactoring the solver and need them).

Is it possible just to comment out all the `#foo# annotations? (I guess there are many?)

Comment (mlutze) [reply main/test/ca/uwaterloo/flix/TestExamples.scala]: There are many. Luckily, I'm becoming a master at writing little scripts to manipulate Flix syntax.

File: main/test/flix/Test.Dec.Class.flix [hunk 2]
```diff
@@ -145,11 +145,11 @@ namespace Test/Dec/Class {
         lawless class C[a] {
             pub def f(): a
 
-            law l(): Bool = true
+            law l: forall (_x: a) . true
 
             pub def g(): a
 
-            law m(): Bool = true
+            law m: forall[a](_x: a) . true
```
Comment (magnus-madsen) [inline main/test/flix/Test.Dec.Class.flix:152]: Does this [a] shadow the a of the class?

**Comment by magnus-madsen** [comment]
Thanks! now to try it out :)

File: main/test/flix/Test.Dec.Class.flix (hunk 2 above)
Comment (mlutze) [reply main/test/flix/Test.Dec.Class.flix:152]: I think it is actually bound to the class's a

**Pull request merged**